### PR TITLE
[DORT] Reduce global configs to make enabling dynamic shape easier

### PR DESCRIPTION
**Pull request by wschin**
There are several global configs used by DORT.
```py
DEFAULT_ONNX_EXPORTER_OPTIONS = torch.onnx._internal.exporter.ResolvedExportOptions(
    torch.onnx._internal.exporter.ExportOptions()
)

# TODO(wechi): This line must generate result identical to the call of
# _create_onnx_supports_op_overload_table(...) inside
# create_onnx_friendly_decomposition_table(...) in
# torch/onnx/_internal/fx/decomposition_table.py.
_SUPPORT_DICT = torch.onnx._internal.fx.decomposition_table._create_onnx_supports_op_overload_table(
    DEFAULT_ONNX_EXPORTER_OPTIONS.onnx_registry
)  # type: ignore

_EXTRA_SUPPORT_DICT: Dict[str, Any] = {
    "getattr": None,
    "_operator.getitem": None,
}

DORT_DECOMPOSITION_TABLE = DEFAULT_ONNX_EXPORTER_OPTIONS.decomposition_table
```

We can see all but `_EXTRA_SUPPORT_DICT` are extracted from deduced from ONNX exporter's options. As there are many ways to configure ONNX exporter's options, we decided to move these variables to `OrtBackend`'s `__init__` so that the construction of `OrtBackend` becomes more flexible (especially for enabling dynamic shape or not).

### CODE REVIEW

File: orttraining/orttraining/python/training/torchdynamo/ort_backend.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 import dataclasses
 import logging
-from typing import Any, Dict, Mapping, Tuple, Union
+from typing import Any, Dict, Mapping, Optional, Set, Tuple, Union
```
Comment (pallavides) [inline orttraining/orttraining/python/training/torchdynamo/ort_backend.py:9]: Consider removing unused imports like torch.onnx._onnx_suported_ops

**Review by pallavides — APPROVED** [review]
Looks good overall, added a few suggestions, thanks!

File: orttraining/orttraining/python/training/torchdynamo/ort_backend.py [hunk 2]
```diff
@@ -115,8 +89,13 @@ class OrtOperatorSupport(OperatorSupport):
...
 
-    def __init__(self):
-        super().__init__(_EXTRA_SUPPORT_DICT)
+    def __init__(self, support_dict: Set[Any], extra_support_dict: Dict[str, Any]):
+        # Use extra_support_dict[op_name] = None to indicate
+        # we support op_name with all input types. Otherwise,
+        # see support_dict (type: SupportDict) in operator_support.py
+        # for specifying supported types.
+        super().__init__(extra_support_dict)
+        self._support_dict = support_dict
```
[the remark this answers is not in the document]
Comment (wschin) [reply orttraining/orttraining/python/training/torchdynamo/ort_backend.py:98]: It's kind of warning to myself --- we don't know if it should be public. use it at your own risk. :)

File: orttraining/orttraining/python/training/torchdynamo/ort_backend.py [hunk 3]
```diff
@@ -390,8 +369,44 @@ class OrtBackend:
...
+            # This is identical to self.resolved_onnx_exporter_options.onnxfunction_dispatcher.onnx_registry.
+            self.resolved_onnx_exporter_options.onnx_registry
+        )  # type: ignore
+
+        extra_support_dict: Dict[str, Any] = {
+            "getattr": None,
+            "_operator.getitem": None,
+        }
+
+        self._supported_ops = OrtOperatorSupport(support_dict, extra_support_dict)
```
[the remark this answers is not in the document]
Comment (wschin) [reply orttraining/orttraining/python/training/torchdynamo/ort_backend.py:409]: It's for being conservative when designing new things. Feel free to change it.

**Pull request merged**